examples/OSPolicyAssignments/console/win-restore-rdp-powershell.yaml

```diff
@@ -0,0 +1,261 @@
+            {
+                $RDP_Default_Sec_Status = $false
+            
+                try
+                {
+                    Write-Host "Checking if Terminal Service is using default RDP security settings.."   
+                    $value = Get-ItemProperty -Path "HKLM:\System\CurrentControlSet\Control\Terminal Server\WinStations\RDP-Tcp" -Name SecurityLayer | Select -ExpandProperty SecurityLayer
+                    if ($value -eq 1){
+                        $RDP_Default_Sec_Status = $true
+                    }                     
+                }
+                catch [System.Exception]
+                {
+                    throw "Error getting registry value for RDP security: $_"
+                }
+                return $RDP_Default_Sec_Status
+            }
+            
+            function Get-RDPDefaultNLA ()
+            {
+                $RDP_Default_NLA_Status = $false
+            
+                try
+                {
+                    Write-Host "Checking if Terminal Service is using default NLA settings.."  
+                    $value = Get-ItemProperty -Path "HKLM:\System\CurrentControlSet\Control\Terminal Server\WinStations\RDP-Tcp" -Name UserAuthentication | Select -ExpandProperty UserAuthentication
+                    if ($value -eq 0){
+                        $RDP_Default_NLA_Status = $true
+                    }                     
+                }
+                catch [System.Exception]
+                {
+                    throw "Error getting registry value for RDP NLA setting: $_"
+                }
+                return $RDP_Default_NLA_Status
+            }
+            
+            function Get-NetworkAdapterDHCP ()
+            {             
+                $Network_Adapter_DHCP_Status = $false
+            
+                try
+                {
+                    Write-Host "Checking if network interface is set to DHCP mode.."  
+                    $value = Get-NetIPInterface -InterfaceAlias Ethernet -AddressFamily IPv4 | Select -ExpandProperty Dhcp
+                    if ($value -eq 'Enabled'){
+                        $Network_Adapter_DHCP_Status = $true
+                    }
+                }
+                catch [System.Exception]
+                {
+                    throw "Error retrieving Network adapter settings: $_"
+                }
+                return $Network_Adapter_DHCP_Status
+            }
+            
+            function Get-GCEInstanceRDPsettings()
+            {
+                Write-Host "Running diagnostic tests.."
+            
+                $r1 = Get-RDPService
+                $r2 = Get-FirewallSettings
+                $r3 = Get-RDPConnections
+                $r4 = Get-RDPDefaultSecurityNeg
+                $r5 = Get-RDPDefaultNLA
+                $r6 = Get-NetworkAdapterDHCP
+                
+                if (($r1 -and $r2 -and $r3 -and $r4 -and $r5 -and $r6) -eq $true)
```

Comment:
This should give the same result:

`if ($r1 -and $r2 -and $r3 -and $r4 -and $r5 -and $r6)`